care/facility/api/serializers/patient.py

```diff
@@ -360,10 +360,18 @@ def update(self, instance, validated_data):
                 Disease.objects.bulk_create(diseases, ignore_conflicts=True)
 
             if meta_info:
-                for key, value in meta_info.items():
-                    setattr(patient.meta_info, key, value)
+                if patient.meta_info is None:
+                    meta_info_obj = PatientMetaInfo.objects.create(**meta_info)
+                    patient.meta_info = meta_info_obj
+                else:
+                    for key, value in meta_info.items():
+                        setattr(patient.meta_info, key, value)
                 patient.meta_info.save()
 
+            else:
+                #patient.save()
+                if patient.meta_info:
+                    patient.meta_info.delete()
```

Comment:
Instead of deleting the meta object just set the valuse to none

https://www.django-rest-framework.org/api-guide/serializers/#writable-nested-representations